packages/webdriverio/src/utils.js

```diff
@@ -5,147 +5,153 @@ import rgb2hex from 'rgb2hex'
 
-    /**
-     * check if user has specified locator strategy directly
-     */
-    const match = value.match(DIRECT_SELECTOR_REGEXP)
-    if (match) {
-        /**
-         * ensure selector strategy is supported
-         */
-        if (!isMobile && isW3C && !W3C_SELECTOR_STRATEGIES.includes(match[1])) {
-            throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
-        }
-
-        return {
-            using: match[1],
-            value: match[2]
-        }
-    }
-
-    // use xPath strategy if value starts with //
-    if (value.indexOf('/') === 0 || value.indexOf('(') === 0 ||
-               value.indexOf('../') === 0 || value.indexOf('./') === 0 ||
-               value.indexOf('*/') === 0) {
-        using = 'xpath'
-
-    // use link text strategy if value starts with =
-    } else if (value.indexOf('=') === 0) {
-        using = 'link text'
-        value = value.slice(1)
-
-    // use partial link text strategy if value starts with *=
-    } else if (value.indexOf('*=') === 0) {
-        using = 'partial link text'
-        value = value.slice(2)
-
-    // recursive element search using the UiAutomator library (Android only)
-    } else if (value.indexOf('android=') === 0) {
-        using = '-android uiautomator'
-        value = value.slice(8)
-
-    // recursive element search using the UIAutomation library (iOS-only)
-    } else if (value.indexOf('ios=') === 0) {
-        using = '-ios uiautomation'
-        value = value.slice(4)
-
-    // recursive element search using accessibility id
-    } else if (value.indexOf('~') === 0) {
-        using = 'accessibility id'
-        value = value.slice(1)
-
-    // class name mobile selector
-    // for iOS = UIA...
-    // for Android = android.widget
-    } else if (value.slice(0, 3) === 'UIA' || value.slice(0, 15) === 'XCUIElementType' || value.slice(0, 14).toLowerCase() === 'android.widget') {
-        using = 'class name'
-
-    // use tag name strategy if value contains a tag
-    // e.g. "<div>" or "<div />"
-    } else if (value.search(/<[a-zA-Z-]+( \/)*>/g) >= 0) {
-        using = 'tag name'
-        value = value.replace(/<|>|\/|\s/g, '')
-
-    // use name strategy if value queries elements with name attributes for JSONWP
-    // or if isMobile is used even when w3c is used
-    // e.g. "[name='myName']" or '[name="myName"]'
-    } else if (isMobile
-        ? isNameAttribute
-        : !isW3C && isNameAttribute) {
-        using = 'name'
-        value = value.match(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/)[2]
-
-    // allow to move up to the parent or select current element
-    } else if (value === '..' || value === '.') {
-        using = 'xpath'
-
-    // any element with given class, id, or attribute and content
-    // e.g. h1.header=Welcome or [data-name=table-row]=Item or #content*=Intro
+    if (isPlainObject(value)) {
+        using = '-android datamatcher'
+        value = JSON.stringify(value)
```

Comment:
```suggestion
        value = JSON.stringify(value)
        
        return { using, value }
```

Since we don't do much else let's return immediately and save indentations.